test/unit-tests.c

```diff
@@ -1684,7 +1684,157 @@ test_dirty_pages_without_dma(UNUSED void **state)
     assert_int_equal(0xabcd, r);
 }
 
-int main(void)
+static void
+test_device_set_irqs(UNUSED void **state)
+{
+    vfu_irqs_t *irqs = alloca(sizeof (*irqs) + sizeof (int));
+    struct vfio_irq_set irq_set = { 0, };
+    vfu_ctx_t vfu_ctx = { 0, };
+    int fd = 0xdead;
+    int ret;
+
+    vfu_ctx.irq_count[VFU_DEV_MSIX_IRQ] = 2048;
+    vfu_ctx.irq_count[VFU_DEV_ERR_IRQ] = 1;
+    vfu_ctx.irq_count[VFU_DEV_REQ_IRQ] = 1;
+    vfu_ctx.irqs = irqs;
+
+    /* validation tests */
+
+    irq_set.argsz = sizeof (irq_set);
+
```

Comment:
It would be helpful to add comments explaining why it's expected to fail, e.g. for the following call to `handle_device_set_irqs` we can have:
```
/* size (0) != irq_set.argsz */

Reply:
hmm, yes